server/src/ZetaSqlWrapper.ts

```diff
@@ -212,7 +227,17 @@ export class ZetaSqlWrapper {
       }
       this.registerTable(table);
     }
-    return this.getAstOrError(compiledSql);
+
+    const ast = await this.getAstOrError(compiledSql);
+    if (ast.isOk() && model) {
+      const table = new TableDefinition([model.database, model.schema, model.name]);
+      table.columns = ast.value.resolvedStatement?.resolvedQueryStmtNode?.outputColumnList
+        .filter(c => c.column !== null)
+        .map(c => ZetaSqlWrapper.createSimpleColumn(c.name, c.column?.type ?? null));
```

Comment:
This is duplicated code. Probably we should simplify `analyzeRef` method. Could you try it?